code401challenges/src/main/java/code401challenges/utilities/AnimalShelter.java

```diff
@@ -0,0 +1,37 @@
+package code401challenges.utilities;
+
+import code401challenges.stacksandqueues.Queue;
+
+public class AnimalShelter<T extends Queue<T>> {
+    public Queue<T> dog;
+    public Queue<T> cat;
+
+    public AnimalShelter() {
+        this.dog = new Queue<>();
+        this.cat = new Queue<>();
+    }
+
+    public void enqueue(T animal) {
+        if(animal.equals(dog)) {

```

Comment:
`dog` was assigned a `Queue` so unless the 'thing' being passed into this method is a queue this conditional statement won't work